recipes/opencv/4.x/conanfile.py

```diff
@@ -26,6 +26,7 @@ class OpenCVConan(ConanFile):
         "with_eigen": [True, False],
         "with_webp": [True, False],
         "with_gtk": [True, False],
+        "gtk_version": [2,3],
```

Comment:
I think that this option can be avoided (but properly handle `WITH_GTK_2_X` option in `build()` is required).

Let me explain:
- this recipe should detect in `build()` which major version of GTK is used in opencv dependency graph, and set `WITH_GTK_2_X` accordingly.
- unless `default_package_id_mode` is `unrelated_mode`, major versions of dependencies always contribute to package_id.
  So if you already have an opencv package with GTK 2.x, update to GTK 3.x in your dependency graph, conan will properly build an other opencv package.

Reply:
Hi @SpaceIm. Thanks for your explanations. I will try to get the current GTK version from the system GTK recipe.

Reply:
You can't! system package is related to system package manager, you would need to check every distribution. The correct way here is providing separated GTK packages, according their version.

Reply:
@uilianries Actually you can get gtk version from gtk options in system branch (or from deps_cpp_info in the non-system branch).

Reply:
True, but the only thing you could do with this information is erroring out of it does not match what is wanted

Our regular Conan GTK recipe provides GTK 3 and GTK 4, maybe someone is motivated to add GTK 2 too 😄 

https://github.com/conan-io/conan-center-index/blob/master/recipes/gtk/config.yml

Reply:
@SpaceIm Of course! I totally forgot, and I made that recipe :facepalm: